app/ui/src/app/customizations/api-connector/api-connector.models.ts

```diff
@@ -1,98 +1,64 @@
-import { StringMap, BaseReducerModel, PlatformStore } from '@syndesis/ui/platform';
+import { StringMap, BaseReducerModel, BaseRequestModel, PlatformStore } from '@syndesis/ui/platform';
 import { BaseEntity } from '@syndesis/ui/model';
 
-export interface ApiConnector extends BaseEntity {
-  kind: string;
-  data: ApiConnectorData;
-}
-
-export interface ApiConnectorData extends ApiConnectorValidation {
-  id: string;
-  name: string;
-  description: string;
-  icon: string;
-  fileIcon?: File;
-  properties: ApiConnectorProperties;
-  connectorProperties: any;
+export interface ApiConnectorValidationError {
+  error?: string;
+  message: string;
+  property?: string;
 }
 
-export type ApiConnectors = Array<ApiConnectorData>;
-
-export interface ApiConnectorProperties {
-  specification: {
-    kind: string;
-    displayName: string;
-    required: boolean;
-    type: string;
-    javaType: string;
-    description: string
+export interface ApiConnectorData {
+  id?: string;
+  actionsSummary?: {
```

Comment:
Wow this changed quite a bit.

Reply:
These models are far from being convenient and Frontend/Angular-friendly, but this is the best I could get to map the information we need from the backend in a simple and lean fashion.

Backend and Frontend will have to compromise in a better way. Otherwise this app will not scale.